lua/entities/gmod_wire_expression2/init.lua

```diff
@@ -584,3 +584,51 @@ function MakeWireExpression2(player, Pos, Ang, model, buffer, name, inputs, outp
 	return self
 end
 duplicator.RegisterEntityClass("gmod_wire_expression2", MakeWireExpression2, "Pos", "Ang", "Model", "_original", "_name", "_inputs", "_outputs", "_vars", "inc_files", "filepath")
+
+--------------------------------------------------
+-- Emergency shutdown (beta testing so far)
+--------------------------------------------------
+local average_ram = 0
+local enable = CreateConVar( 
+	"wire_expression2_ram_emergency_shutdown_enable", "0", {FCVAR_ARCHIVE}, 
+	"Enable/disable the emergency shutdown feature. Requires map reload after change." )
+
+local average_halt_multiplier = CreateConVar( 
+	"wire_expression2_ram_emergency_shutdown_spike", "4", {FCVAR_ARCHIVE}, 
+	"if (current_ram > average_ram * spike_convar) then shut down all E2s" )
+
+local halt_max_amount = CreateConVar( 
+	"wire_expression2_ram_emergency_shutdown_total", "512", {FCVAR_ARCHIVE}, 
+	"This is in kilobytes, if (current_ram > total_convar) then shut down all E2s" )
+
+if enable:GetBool() then
+	hook.Remove( "Think", "wire_expression2_emergency_shutdown" ) -- remove old hook
```

Comment:
I have that because if I don't remove the old hook, you end up with 50 of
them during testing (every time you save the file after editing it, it runs
again and creates a new hook).
You could remove it, but it does no harm so I left it.

On Fri, Nov 6, 2015 at 5:44 AM, immibis <notifications@github.com> wrote:

> In lua/entities/gmod_wire_expression2/init.lua
> <https://github.com/wiremod/wire/pull/964#discussion_r44103855>:
>
> > +--------------------------------------------------
> > +local average_ram = 0
> > +local enable = CreateConVar(
> > +	"wire_expression2_ram_emergency_shutdown_enable", "0", {FCVAR_ARCHIVE},
> > +	"Enable/disable the emergency shutdown feature. Requires map reload after change." )
> > +
> > +local average_halt_multiplier = CreateConVar(
> > +	"wire_expression2_ram_emergency_shutdown_spike", "4", {FCVAR_ARCHIVE},
> > +	"if (current_ram > average_ram * spike_convar) then shut down all E2s" )
> > +
> > +local halt_max_amount = CreateConVar(
> > +	"wire_expression2_ram_emergency_shutdown_total", "512", {FCVAR_ARCHIVE},
> > +	"This is in kilobytes, if (current_ram > total_convar) then shut down all E2s" )
> > +
> > +if enable:GetBool() then
> > +	hook.Remove( "Think", "wire_expression2_emergency_shutdown" ) -- remove old hook
>
> You don't need to do this. hook.Add will overwrite old hooks with the same
> name.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wiremod/wire/pull/964/files#r44103855>.
>